lib/simplifyExpression/collectAndCombineSearch/multiplyLikeTerms.js

```diff
@@ -93,51 +87,82 @@ function multiplyPolynomialTerms(node) {
+          Node.Creator.constant(1),
+          polyTerm.getCoeffNode());
+
+        newNode.args[i].changeGroup = changeGroup;
+        node.args[i].changeGroup = changeGroup; // note that this is the "oldNode"
+
+        change = true;
+        changeGroup++;
+      }
+    });
+  }
 
   if (change) {
     return Node.Status.nodeChanged(
-        ChangeTypes.ADD_EXPONENT_OF_ONE, node, newNode, false);
+      ChangeTypes.ADD_EXPONENT_OF_ONE, node, newNode, false);
   }
   else {
     return Node.Status.noChange(node);
   }
 }
-
 // Given a product of polynomial terms, groups the exponents into a sum
 // e.g. x^2 * x^3 * x^1 -> x^(2 + 3 + 1)
 // Returns a Node.Status object.
 function collectExponents(node) {
-  const polynomialTermList = node.args.map(n => new Node.PolynomialTerm(n));
-
-  // If we're multiplying polynomial nodes together, they all share the same
-  // symbol. Get that from the first node.
-  const symbolNode = polynomialTermList[0].getSymbolNode();
-
-  // The new exponent will be a sum of exponents (an operation, wrapped in
-  // parens) e.g. x^(3+4+5)
-  const exponentNodeList = polynomialTermList.map(p => p.getExponentNode(true));
-  const newExponent = Node.Creator.parenthesis(
-    Node.Creator.operator('+', exponentNodeList));
-  const newNode = Node.Creator.polynomialTerm(symbolNode, newExponent, null);
-  return Node.Status.nodeChanged(
-    ChangeTypes.COLLECT_EXPONENTS, node, newNode);
+  if (checks.canMultiplyLikeTermConstantNodes(node)) {
+    const constantTermList = node.args.map(n => new Node.ConstantTerms(n));
+    // If we're multiplying constant nodes together, they all share the same
+    // base. Get that from the first node.
+    const baseNode = constantTermList[0].getBaseNode();
+    // The new exponent will be a sum of exponents (an operation, wrapped in
+    // parens) e.g. 10^(3+4+5)
+    const exponentNodeList = constantTermList.map(p => p.getExponentNode(true));
+    const newExponent = Node.Creator.parenthesis(
+      Node.Creator.operator('+', exponentNodeList));
+    const newNode = Node.Creator.constantTerm(baseNode, newExponent, null);
+    return Node.Status.nodeChanged(
+      ChangeTypes.COLLECT_EXPONENTS, node, newNode);
+  }
+  else {
+    const polynomialTermList = node.args.map(n => new Node.PolynomialTerm(n));
+    // If we're multiplying polynomial nodes together, they all share the same
+    // symbol. Get that from the first node.
+    const symbolNode = polynomialTermList[0].getSymbolNode();
+    // The new exponent will be a sum of exponents (an operation, wrapped in
+    // parens) e.g. x^(3+4+5)
+    const exponentNodeList = polynomialTermList.map(p => p.getExponentNode(true));
+    const newExponent = Node.Creator.parenthesis(
+      Node.Creator.operator('+', exponentNodeList));
+    const newNode = Node.Creator.polynomialTerm(symbolNode, newExponent, null);
+    return Node.Status.nodeChanged(
+      ChangeTypes.COLLECT_EXPONENTS, node, newNode);
```

Comment:
no worries - if things don't work and it gets annoying I'm happy to help troubleshoot